### ISSUE TITLE
Search option for "doesn't work with 4.0"

### ISSUE THREAD

**Issue by sminnee** [silverstripe/addons.silverstripe.org#161]
Currently you can search for module that work with 4.0 but not for modules that don't.

It might be the kind of thing that's buried in an "advanced search" section or even just an optional querystring param that you can add manually, but it would help with things like measuring progress of module upgrades.

**Comment by maxime-rainville** [comment]
I've got a simple proof of concept working in my box. I've implemented it has a `negativeCompatibility` hidden field. So you need to know it's there to use it.

I might need some UI guidance if we want to make this option publicly visible. Otherwise, I can do a PR as is.

**Comment by robbieaverill** [comment]
cc @clarkepaul for UI ideas

**Comment by clarkepaul** [comment]
Yup, I think we could have a field pre-set to v4 by default, and they can set to another version. 

Just taken a look... on the modules sections there is already a section for selecting a version and it seems to work. 
![image](https://user-images.githubusercontent.com/555033/40944626-94e457ba-68a9-11e8-88fd-f935b0319d30.png)

On the landing page of Addons we don't have the advanced section but on the search results we do. Are you referring to searching from silverstripe.org @sminnee?

**Comment by sminnee** [comment]
I'm talking about the search advanced section as shown on search results, as you have screen-grabbed.

The purpose of this card was to allow for search of incompatible version as well as compatible. In particular, this was to let us see which modules have _not_ yet been upgraded to SS4. 

In combination with #165 this could let us search for modules that don't yet have *stable releases* for SS4.

**Comment by sminnee** [comment]
It might be worth considering #165 in any UI design we do, as that would add 1 more option to the search form too.

**Comment by clarkepaul** [comment]
Cool, got it now.
